include/nonkeyedcommand.h

```diff
@@ -0,0 +1,144 @@
+*   \details The KeyedCommand class has multiple methods for dealing
+*            with non-keyed commands.
+*/
+class NonKeyedCommand : public Command
+{
+    public:
+        /*!
+        *   \brief Set address and port for command
+        *          to be executed on
+        *   \param address Address of database
+        *   \param port Port of database
+        */
+        void set_exec_address_port(std::string address,
+                                   uint16_t port);
+
+        /*!
+        *   \brief Get address that command will be
+        *          to be executed on
+        *   \return std::string of address
+        *           if an address hasn't been set,
+        *                 returns an empty string
+        */
+        std::string get_address();
+
+        /*!
+        *   \brief Get port that command will be
+        *          to be executed on
+        *   \return uint16_t of port
+        *           if port hasn't been set, returns 0
+        */
+        uint16_t get_port();
+
+        virtual CommandReply runme(RedisServer * r) = 0;
+
+    private:
+        /*!
+        *   \brief Address of database node
+        */
+        std::string _address;
+
+        /*!
+        *   \brief Port of database node
+        */
+        uint64_t _port;
+};
+
+/*!
+*   \brief The AddressAtCommand class constructs non-keyed Client
+*          commands that address the user given db node(s).
+*          This is a subclass of the NonKeyedCommand class.
+*/
+class AddressAtCommand : public NonKeyedCommand
+{
+    public:
+        virtual CommandReply runme(RedisServer * r);
+};
+
+/*!
+*   \brief The AddressAnyCommand class constructs non-keyed Client
+*          commands that address any active db node(s).
+*          This is a subclass of the NonKeyedCommand class.
+*/
+class AddressAnyCommand : public NonKeyedCommand
+{
+    public:
+        virtual CommandReply runme(RedisServer * r);
+};
+
+class DBInfoCommand : public AddressAtCommand
+{
+    public:
+        virtual CommandReply runme(RedisServer *r);
+
+        /*!
+        *   \brief Parse database node information from get_db_node_info()
+        *          into a nested unordered_map
+        *   \param info containing the database node information
+        *   \return parsed_reply_nested_map containing the database node information
+        */
+        parsed_reply_nested_map parse_db_node_info(std::string info);
+};
+
+class ClusterInfoCommand : public AddressAtCommand
+{
+    public:
+        virtual CommandReply runme(RedisServer *r);
+
+        /*!
+        *   \brief Parse database node information from get_db_node_info()
+        *          into a nested unordered_map
+        *   \param info containing the database node information
+        *   \return parsed_reply_map containing the database node cluster information
+        */
+        parsed_reply_map parse_db_cluster_info(std::string info);
```

Comment:
I wonder if we should make these static member functions so we can call parsing routines without need to create the object, i.e. ``ClusterInfoCommand::parse_db_cluster_info()``